### ISSUE TITLE
Luminance model of chrominance filter switches red and blue values

### ISSUE THREAD

**Issue by elanvrt** [CasparCG/Server#174]
I tested the CasparCG chrominance filter on my testcard, and what I saw surprised me: In the luminance model (Y), the red turns too dark and the blue too bright, as in the following picture, captured from CasparCG in PAL mode, with zero chrominance:
![casparcg_sat0](https://f.cloud.github.com/assets/4322556/1210245/38a4ae2c-25f7-11e3-8669-a87a12feb541.jpg)
This accords to the CCIR709 colour space, except that the values for R and B are switched: Y = 0.073R + 0.72G + 0.21B. 

THIS IS OBVIOUSLY A BUG. 

In addition, for SDTV, the CCIR601 luminance model should be used to obtain a correct monochrome, as in following picture: (not made with CasparCG)
![ccir601_mono](https://f.cloud.github.com/assets/4322556/1210247/3c0f342e-25f7-11e3-9720-39eae210adc5.jpg)

So normally, the luminance value for HDTV should be: 
Y = 0.21R + 0.715G + 0.072B (CCIR709)
and for PAL/NTSC: 
Y= 0.299R + 0.587G + 0.114B (CCIR601)

For reference: This is the colour picture I used:
![elan169](https://f.cloud.github.com/assets/4322556/1210369/b267533a-25f8-11e3-9afd-4378dc5de2b3.jpg)

Putting CasparCG in HD 1080i50 mode made no difference when using the same test card.

**Comment by HellGore** [comment]
I could not find a corresponding testpattern for HD but generated one from this file via the FFmpeg filter

```
scale=in_color_matrix=bt709:out_color_matrix=bt601:w=1920:h=1080
```

I could not figure out why I needed to reverse the input and output color matrix though.

I ended up with the following full hd image:

![testbild_hd](https://cloud.githubusercontent.com/assets/2980939/20610283/a7aa7646-b297-11e6-878c-7c416d45fec3.png)

Could somebody verify that my image is indeed correct from a BT.709 standpoint?

**Comment by HellGore** [comment]
@elanvrt could you give me some files to test with that demonstrates that this is **not** fixed?

**Comment by elanvrt** [comment]
Actually, it does not matter which colorbar picture you use. the one I supplied is for CCIR601 PAL but the colorbar pics supplied with casparCG also show the same error: The blue bar is brighter than the red bar when MIXER SATURATION is set to 0 (or a very low value). The actual brightness differs with colorspace, but any error in colorspace is not so bad as this luminance reversal between red and blue. The bug must definitely be sought in the code that processes MIXER SATURATION.

**Comment by HellGore** [comment]
I did a channel snapshot and I got 
![saturation_0](https://cloud.githubusercontent.com/assets/2980939/22737960/b23049ac-ee05-11e6-918f-fdc95e745689.png)
In what way do you mean that it is incorrect? Or are you accidentally testing against 2.0.7 or something?

**Comment by HellGore** [comment]
Actually now I'm absolutely sure why you still get the problem -- This was fixed **after** The 2.1.0 Beta 1 release, and you had the problem in 2.1.0 Beta 1. You need to try a newer nightly build.

**Comment by elanvrt** [comment]
Indeed, so we may finally close this issue. Thanks for the fix. Over & out.
